app/controllers/admin/tag_sets_controller.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+##
+# This class is the controller for Tag Sets which are used to link together two related tag groups.
+# It allows you to create and view Tag Sets.
+class Admin::TagSetsController < ApplicationController
+  authorize_resource
+
+  def index
+    @tag_sets = TagSet.all
+
+    respond_to { |format| format.html }
+  end
+
+  def show
+    @tag_set = TagSet.find(params[:id])
+
+    respond_to { |format| format.html }
+  end
+
+  def new
+    @tag_set = TagSet.new
+
+    respond_to { |format| format.html }
+  end
+
+  def create
+    @tag_set = TagSet.new(tag_set_params)
+
+    respond_to do |format|
+      if @tag_set.save
+        flash[:notice] = 'Tag Set was successfully created.'
+        format.html { redirect_to admin_tag_set_path(@tag_set) }
+      else
+        format.html { render action: :new }
```

Comment:
Looks like when save returns false (I assume that's bad) it just returns to the new action?

Do you think we should respond that the previous response was erroneous or something?

Reply:
I don't know if this is what's happening here, but this is usual Rails behaviour.  Save returning false also sets errors on a list of errors on the object.  If the `new` page is set up correctly, it should render any values in that error list so that the first time you go to the page, it doesn't show any.  And if a previous attempt occurred, you see the list of issues generated last time a save was attempted.